name: Issue Related Tests

on:
  pull_request:
    types: [opened, edited, synchronize]

jobs:
  issue-related-tests:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2

    - name: Set up Python
      uses: actions/setup-python@v2
      with:
        python-version: '3.12.3'

    - name: Cache pipenv virtualenv
      uses: actions/cache@v2
      with:
        path: ~/.local/share/virtualenvs
        key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}

    - name: Install pipenv
      run: |
        pip install pipenv
      continue-on-error: true

    - name: Install dependencies
      run: |
        pipenv install --dev
      continue-on-error: true

    - name: Set PYTHONPATH and check environment
      run: |
        echo "Current working directory: $(pwd)"
        ls -la
        export PYTHONPATH=$PYTHONPATH:$(pwd)
        echo "PYTHONPATH: $PYTHONPATH"
        pipenv --venv
      continue-on-error: true

    - name: Run API script
      run: |
        pipenv run python -m core/utils/issue_related_tests.py
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        REPO: ${{ github.repository }}
        OWNER: ${{ github.repository_owner }}
        BRANCH: ${{ github.head_ref }}
        DJANGO_SETTINGS_MODULE: 'REsQue.settings'
      continue-on-error: true

    - name: Check for script errors
      if: failure()
      run: |
        echo "The script failed to run. Please check the logs for more information."
        exit 1